### PR TITLE
Add solutions in Ruby and Bash (Linux/Unix)

### PR DESCRIPTION
**Pull request by LyubomirT**
This pull request adds implementations for the file hash comparison utility in both Ruby and Bash. As requested in #1. Tried to replicate the code written in JavaScript and Go fully.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  def initialize
+    @storage = {}
+  end
+
+  def add_hash(hash, file_path)
+    if @storage[hash]
+      @storage[hash] << file_path
+    else
+      @storage[hash] = [file_path]
+    end
```
Comment (kerolloz) [inline main.rb]: can we rewrite that into a shorthand if else

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  hash_storage = HashStorage.new
+  read_dir(dir).each do |file|
+    hash_storage.add_hash(get_file_hash(file), file)
+  end
+  puts hash_storage.get_similar.inspect
+  exit(0)
+else
+  puts "Please provide a valid directory path"
+  exit(-1)
+end
```
Comment (kerolloz) [inline main.rb]: let's make that as a guard check, so maybe a variable named `is_valid_dir = dir && Dir.exist?(dir)` and then do the check `abort("Invalid path") if not is_valid_dir`

**Review by kerolloz — COMMENTED** [review]
Thanks a lot for your PR ❤️

File: main.rb (hunk 1 above)
Comment (LyubomirT) [reply main.rb]: Applied these changes.

File: main.rb (hunk 2 above)
Comment (LyubomirT) [reply main.rb]: Applied these changes.

**Comment by LyubomirT** [comment]
@kerolloz Please review this PR again, I've just applied the necessary changes.

**Comment by kerolloz** [comment]
Merged. Thank you @LyubomirT

**Pull request merged**